driver/ssh/config.go

```diff
@@ -27,14 +28,18 @@ func (conf *Config) EncodeWithDSN(base string) (string, error) {
 	updatedBase := parsed.String()
 	split := strings.Split(updatedBase, "://")
 
+	if len(split) < 2 {
+		return "", errors.New("invalid dsn format to read the warehouse dsn")
+	}
+
 	return fmt.Sprintf(
 		"%s://%s@%s:%d/%s", split[0], conf.User, conf.Host, conf.Port, split[1]), nil
 }
 
 func (conf *Config) DecodeFromDSN(encodedDSN string) (dsn string, err error) {
 	parsed, err := url.Parse(encodedDSN)
 	if err != nil {
-		return "", fmt.Errorf("parsing the encoded dsn: %s, %s", encodedDSN, err.Error())
+		return "", fmt.Errorf("parsing the encoded DSN: %s", err.Error())
```

Comment:
Can we stick to using `%v` or `%w` and just pass `err`? Not important but there is no much point in calling `err.Error()` and it could protect us against `err` being `nil` as well.

```suggestion
		return "", fmt.Errorf("parsing the encoded DSN: %v", err)
```